ddl/ddl.go

```diff
@@ -1157,16 +1494,132 @@ func IterAllDDLJobs(txn kv.Transaction, finishFn func([]*model.Job) (bool, error
 	if err != nil || finish {
 		return err
 	}
-	return IterHistoryDDLJobs(txn, finishFn)
+	return IterHistoryDDLJobs(ctx, txn, finishFn)
+}
+
+// GetLastNHistoryDDLJobs get last n history ddl jobs.
+func GetLastNHistoryDDLJobs(sess *session, m *meta.Meta, maxNumJobs int) ([]*model.Job, error) {
+	if variable.EnableConcurrentDDL.Load() {
+		return getJobsBySQL(sess, "tidb_ddl_history", "1 order by job_id desc limit "+strconv.Itoa(maxNumJobs))
+	}
+	jobs, err := m.GetLastNHistoryDDLJobs(maxNumJobs)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return jobs, nil
+}
+
+// GetLastHistoryDDLJobsIterator gets latest N history ddl jobs iterator.
+func GetLastHistoryDDLJobsIterator(sess sessionctx.Context, m *meta.Meta) (meta.LastJobIterator, error) {
+	if variable.EnableConcurrentDDL.Load() {
+		return &ConcurrentDDLLastJobIterator{
+			sess:   newSession(sess),
+			offset: 0,
+		}, nil
+	}
+	return m.GetLastHistoryDDLJobsIterator()
+}
+
+// ConcurrentDDLLastJobIterator is the iterator for gets latest history.
+type ConcurrentDDLLastJobIterator struct {
+	sess   *session
+	offset uint64
+}
+
+// GetLastJobs iter latest num history ddl jobs.
+func (c *ConcurrentDDLLastJobIterator) GetLastJobs(num int, _ []*model.Job) ([]*model.Job, error) {
+	jobs, err := getJobsBySQL(c.sess, "tidb_ddl_history", fmt.Sprintf("1 order by job_id desc limit %d, %d", c.offset, num))
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	c.offset += uint64(len(jobs))
+	return jobs, err
+}
+
+// session wraps sessionctx.Context for transaction usage.
+type session struct {
+	s sessionctx.Context
```

Comment:
```suggestion
	sessionctx.Context
```
It feels better to change it this way, and then we can also get rid of the 'session()' function